src/utils/amounts.ts

```diff
@@ -1,14 +1,28 @@
-interface TokenAmountToFiatParams {
-  amount: number;
-  token: "eth" | "dai";
-}
+import { BigNumber } from "ethers";
+import { Token } from "src/domain";
 
 const rates = {
   dai: 1.02,
   eth: 2500,
 };
 
-export const convertTokenAmountToFiat = ({ amount, token }: TokenAmountToFiatParams): string => {
+export const convertTokenAmountToFiat = ({
+  amount,
+  token,
+}: {
+  amount: number;
+  token: "eth" | "dai";
+}): string => {
   const fiat = amount * rates[token];
   return `$${fiat.toFixed(2)}`;
 };
+
+export const tokenAmountToNumber = ({
+  amount,
+  token,
+}: {
+  amount: BigNumber;
+  token: Token;
+}): number => {
+  return Number(amount.toString()) / Math.pow(10, token.decimals);
```

Comment:
We should never convert `BigNumber` to `Number` as it can throw an overflow error.

We don't need this function as a util as well, as `ethers` already have one for this as a util that you can directly use in the components:

```ts
ethers.utils.formatUnits(amount, token.decimals);
```

Reply:
So we'll work  with strings or Bignumbers and never numbers